### PR TITLE
op.c: prevent -Wconversion warnings from the opslab code

### PR DESCRIPTION
**Pull request by tonycoz**
One little step.

### CODE REVIEW

**Comment by jkeenan** [comment]
1. With what `./Configure` options or on what platforms are you seeing these warnings?
2. Note one failing CI test on a `g++` build.

**Comment by tonycoz** [comment]
> 1. With what `./Configure` options or on what platforms are you seeing these warnings?

Add `-Accflags=-Wconversion` on Linux - note that this will produce many warnings, this change only fixes a small number of them.

>     2. Note one failing CI test on a `g++` build.

I'll look into it.

**Comment by jkeenan** [comment]
> One little step.

Indeed!

**Before:**
```
$ gitcurr
blead
$ sh ./Configure -des -Dusedevel -Accflags=-Wconversion 
...
$ report-build-warnings 6be4da51a4.linux.conversion.gcc-11.maketp.output.txt.gz
File:  6be4da51a4.linux.conversion.gcc-11.maketp.output.txt.gz

  Wconversion                              1277
  Wfloat-conversion                         11
  Wsign-conversion                         7087
```
**After:**
```
$ gitcurr
tonycoz-wconversion-opslab
$ sh ./Configure -des -Dusedevel -Accflags=-Wconversion 
...
$ report-build-warnings c3c50423c0.linux.conversion.gcc-11.tonyc.maketp.output.txt.gz
File:  c3c50423c0.linux.conversion.gcc-11.tonyc.maketp.output.txt.gz

  Wconversion                              1261
  Wfloat-conversion                         11
  Wsign-conversion                         7087

```

File: op.c [hunk 1]
```diff
@@ -207,11 +207,18 @@ Perl_op_prune_chain_head(OP** op_p)
 #endif
 
 /* rounds up to nearest pointer */
-#define SIZE_TO_PSIZE(x)	(((x) + sizeof(I32 *) - 1)/sizeof(I32 *))
+PERL_STATIC_INLINE U16
+S_size_to_psize(size_t sz) {
+    size_t psize = (sz + sizeof(I32 *) - 1) / sizeof(I32 *);
+    assert(psize < U16_MAX);
```
Comment (mauke) [inline op.c]: Why `<` and not `<=`?

Comment (tonycoz) [reply op.c]: I don't remember.

I don't think it matters much here since the default maximum slab size is 2048.

**Comment by jkeenan** [comment]
@tonycoz do you wish to move forward with this pull request? @mauke, do you want changes?  Thanks.

File: op.c [hunk 2]
```diff
@@ -207,11 +207,30 @@ Perl_op_prune_chain_head(OP** op_p)
...
+  Find the U16 offset of an OPSLOT within an OPSLAB.
+
+*/
+
+PERL_STATIC_INLINE U16
+S_opslab_slot_offset(const OPSLAB *slab, const OPSLOT *slot) {
+    PERL_ARGS_ASSERT_OPSLAB_SLOT_OFFSET;
+
+    const char *base = (const char *)&slab->opslab_slots;
+    assert((size_t)((const char *)(slot) - base) % sizeof(I32**) == 0);
```
Comment (mauke) [inline op.c]: We don't need the parens in `(slot)` now that we're not a macro anymore.

Comment (tonycoz) [reply op.c]: Fixed.

**Review by khwilliamson — APPROVED** [review]

**Pull request merged**